### PR TITLE
More Intellisense Stuffs

### PR DESCRIPTION
**Pull request by MrRowey**
Mainly Files from AI\OpAI with a few other files from lua\

### CODE REVIEW

File: lua/AI/aiattackutilities.lua [hunk 1]
```diff
@@ -411,6 +394,13 @@ function GetBestThreatTarget(aiBrain, platoon, bSkipPathability)
 
 end
 
+---@param aiBrain AIBrain
+---@param platoon Platoon
+---@param location string
```
Comment (Garanas) [inline lua/AI/aiattackutilities.lua:399]: Vector

File: lua/AI/aiattackutilities.lua [hunk 2]
```diff
@@ -532,18 +519,15 @@ function GetNavalPlatoonMaxRange(aiBrain, platoon)
...
---       Finds if the platoon can move to the location given, or close enough to bombard
---   Returns:
---       bool
--------------------------------------------------------
+--- ## Function: CheckNavalPathing
+--- ### Description:
+--- Finds if the platoon can move to the location given, or close enough to bombard
+---@param aiBrain AIBrain aiBrain to use
+---@param platoon Platoon platoon to find best target for
+---@param location string spot we want to get to
```
Comment (Garanas) [inline lua/AI/aiattackutilities.lua:527]: Vector

File: lua/AI/aiattackutilities.lua [hunk 3]
```diff
@@ -612,16 +596,12 @@ function CheckNavalPathing(aiBrain, platoon, location, maxRange, selectedWeaponA
...
---       Gets the path to a random naval marker.
---   Returns:
---       A table representing the path
--------------------------------------------------------
+--- ## Function: AINavalPlanB
+--- ### Description:
+--- Gets the path to a random naval marker.
+---@param aiBrain AIBrain aiBrain to use
+---@param platoon Platoon platoon to find best target for
+---@return table[] # A table representing the path
```
Comment (Garanas) [inline lua/AI/aiattackutilities.lua:604]: Vector[]

File: lua/AI/aiattackutilities.lua [hunk 4]
```diff
@@ -822,21 +791,18 @@ function AIPlatoonSquadAttackVector(aiBrain, platoon, bAggro)
...
---       true if successful, false if couldn't use transports
--------------------------------------------------------
 
+--- ## Function: SendPlatoonWithTransports
+--- ### Description:
+--- Find transports and use them to move platoon.  If bRequired is set, then have platoon
+--- wait 60 seconds for transports before failing
+---@param aiBrain AIBrain aiBrain to use
+---@param platoon Platoon platoon to find best target for
+---@param destination string table representing the destination location
```
Comment (Garanas) [inline lua/AI/aiattackutilities.lua:801]: Vector

File: lua/AI/aiattackutilities.lua [hunk 5]
```diff
@@ -1008,6 +974,12 @@ function SendPlatoonWithTransports(aiBrain, platoon, destination, bRequired, bSk
     return true
 end
 
+---@param aiBrain AIBrain
+---@param platoon Platoon
+---@param destination string
```
Comment (Garanas) [inline lua/AI/aiattackutilities.lua:979]: Vector

File: lua/AI/aiattackutilities.lua [hunk 6]
```diff
@@ -1238,22 +1203,20 @@ function GetMostRestrictiveLayer(platoon)
...
--------------------------------------------------------
-
+--- ## Function: PlatoonGenerateSafePathTo
+--- ### Description:
+--- If there are pathing nodes available to this platoon's most restrictive movement type, then a path to the destination
+--- can be generated while avoiding other high threat areas along the way.
+---@param aiBrain AIBrain aiBrain to use
+---@param platoonLayer Platoon layer to use to generate safe path... e.g. 'Air', 'Land', etc.
+---@param start table table representing starting location
+---@param destination table table representing the destination location
```
Comment (Garanas) [inline lua/AI/aiattackutilities.lua:1213]: Vector

File: lua/AI/aiattackutilities.lua [hunk 7]
```diff
@@ -1238,22 +1203,20 @@ function GetMostRestrictiveLayer(platoon)
...
---       a table of locations representing the safest path to get to the specified destination
--------------------------------------------------------
-
+--- ## Function: PlatoonGenerateSafePathTo
+--- ### Description:
+--- If there are pathing nodes available to this platoon's most restrictive movement type, then a path to the destination
+--- can be generated while avoiding other high threat areas along the way.
+---@param aiBrain AIBrain aiBrain to use
+---@param platoonLayer Platoon layer to use to generate safe path... e.g. 'Air', 'Land', etc.
+---@param start table table representing starting location
```
Comment (Garanas) [inline lua/AI/aiattackutilities.lua:1212]: Vector

File: lua/AI/aiattackutilities.lua [hunk 8]
```diff
@@ -1353,19 +1311,13 @@ function GetPathGraphs()
...
---   Description:
---       Gets the name of the closest pathing node (within radius distance of location) on the layer we specify.
---   Returns:
---        Closest pathing node's name else false
--------------------------------------------------------
-
+--- ## Function: GetClosestPathNodeInRadiusByLayer
+--- ### Description:
+--- Gets the name of the closest pathing node (within radius distance of location) on the layer we specify.
+---@param location string location to search around
```
Comment (Garanas) [inline lua/AI/aiattackutilities.lua:1317]: Vector

File: lua/AI/aiattackutilities.lua [hunk 9]
```diff
@@ -1389,18 +1341,13 @@ function GetClosestPathNodeInRadiusByLayer(location, radius, layer)
...
---   Description:
---       If there is a node from a specific graph within radius distance of location, this function will get its name.
---   Returns:
---       The closest node's name else false
--------------------------------------------------------
-
+--- ## Function: GetClosestPathNodeInRadiusByGraph
+--- ### Description:
+--- If there is a node from a specific graph within radius distance of location, this function will get its name.
+---@param location string location to search around
```
Comment (Garanas) [inline lua/AI/aiattackutilities.lua:1347]: Vector

File: lua/AI/aiattackutilities.lua [hunk 10]
```diff
@@ -1576,6 +1518,14 @@ function GeneratePathSorian(aiBrain, startNode, endNode, threatType, threatWeigh
     return queue
 end
 
+---@param aiBrain AIBrain
+---@param startNode any
+---@param endNode any
+---@param threatType any
+---@param threatWeight any
+---@param endPos any
```
Comment (Garanas) [inline lua/AI/aiattackutilities.lua:1526]: Vector

File: lua/AI/aiattackutilities.lua [hunk 11]
```diff
@@ -1576,6 +1518,14 @@ function GeneratePathSorian(aiBrain, startNode, endNode, threatType, threatWeigh
...
+---@param startPos any
```
Comment (Garanas) [inline lua/AI/aiattackutilities.lua:1527]: Vector

File: lua/AI/aiattackutilities.lua [hunk 12]
```diff
@@ -1576,6 +1518,14 @@ function GeneratePathSorian(aiBrain, startNode, endNode, threatType, threatWeigh
     return queue
 end
 
+---@param aiBrain AIBrain
+---@param startNode any
+---@param endNode any
+---@param threatType any
+---@param threatWeight any
```
Comment (Garanas) [inline lua/AI/aiattackutilities.lua:1525]: Number

**Review by Garanas — CHANGES_REQUESTED** [review]
All is fine when all conversations are resolved.

**Comment by speed2CZ** [comment]
You can delete all unused imports in OpAI folder

**Comment by MrRowey** [comment]
As long as jip okays this i can make another pass in removing them as all unused imports I've been told to keep and stick at the bottom

**Comment by Garanas** [comment]
That is fine - that is the campaign AI. I do not recall a single mod that works on top of those files

**Review by Garanas — APPROVED** [review]

**Pull request merged**